src/dbnode/storage/shard.go

```diff
@@ -385,8 +385,12 @@ func (s *dbShard) RetrievableBlockColdVersion(blockStart time.Time) (int, error)
 // BlockStatesSnapshot implements series.QueryableBlockRetriever
 func (s *dbShard) BlockStatesSnapshot() series.ShardBlockStateSnapshot {
 	s.RLock()
+	defer s.RUnlock()
+	return s.blockStatesSnapshotWithRLock()
```

Comment:
is it safe to acquire the flushState RLock while holding the shard RLock?

Reply:
Yep, I checked usages of the flushState Rlock and they don't conflict have a relationship anywhere else in the code, so this should be good.